### PR TITLE
Enhance MongoDB Dump Restoration Script with Error Handling and Configuration Updates

### PR DESCRIPTION
**Pull request by TeachMeTW**
#### Description:
This pull request refactors the existing MongoDB dump restoration script to include additional features and improvements. The updated script now provides better error handling, debugging information, and integrates with a Docker Compose configuration file to dynamically update the database host configuration.

#### Changes:
1. **Script Header**:
   - Added `#!/bin/bash` shebang to specify the script interpreter.

2. **Directory and Configuration File**:
   - Determined the script directory and configuration file path dynamically.
   - Configured the path to `docker-compose-dev.yml` one level up from the script's directory.

3. **Argument Validation**:
   - Added checks to ensure exactly one argument (the path to the MongoDB dump file) is provided.
   - Added usage instructions and error handling if the argument is missing.

4. **File Existence Checks**:
   - Verified the existence of the provided MongoDB dump file.
   - Verified the existence of the Docker Compose configuration file.

5. **Configuration File Details**:
   - Outputted details about the Docker Compose configuration file for debugging.

6. **Database Name Extraction**:
   - Extracted the database name from the MongoDB dump file to update the Docker Compose configuration file accordingly.

7. **Configuration File Update**:
   - Updated the `DB_HOST` in the Docker Compose configuration file with the actual database name.

8. **Docker Operations**:
   - Printed debug information about the Docker operations.
   - Copied the MongoDB dump file to the Docker container.
   - Cleared existing databases in the Docker container, except for `admin` and `local`.
   - Restored the MongoDB dump to the appropriate database.

9. **Temporary Directory Cleanup**:
   - Removed the temporary directory created for extracting the dump file.

### CODE REVIEW

File: docker/load_mongodump.sh [hunk 1]
```diff
@@ -1,10 +1,70 @@
...
+cat "$CONFIG_FILE"
+
+echo "Copying file to Docker container"
+docker cp "$MONGODUMP_FILE" op-admin-dashboard-db-1:/tmp
+
+FILE_NAME=$(basename "$MONGODUMP_FILE")
 
-FILE_NAME=`basename $MONGODUMP_FILE`
+echo "Clearing existing database"
+docker exec op-admin-dashboard-db-1 bash -c 'mongo --eval "db.getMongo().getDBNames().forEach(function(d) { if (d !== \"admin\" && d !== \"local\") db.getSiblingDB(d).dropDatabase(); })"'
```
Comment (shankari) [inline docker/load_mongodump.sh:65]: we now already have DB_NAME, so I don't think you need to use `getDBNames`, right?
Can't you just do something like:

```
mongo $DB_NAME --eval "db.dropDatabase()"
```

from https://stackoverflow.com/a/8857373

**Review by shankari — CHANGES_REQUESTED** [review]
Almost there, just want to simplify it a bit.

File: docker/load_mongodump.sh (hunk 1 above)
Comment (TeachMeTW) [reply docker/load_mongodump.sh]: @shankari You're correct, I seemed to have overlooked that. It should be good now?

**Review by shankari — APPROVED** [review]
LGTM!

**Comment by shankari** [comment]
🥳 Congratulations on the first merge! 🎉

**Comment by shankari** [comment]
@iantei can you make a similar change to the public dashboard as well?

**Comment by iantei** [comment]
Sure, I will make similar changes to the public dashboard.

File: docker-compose-dev.yml [hunk 2]
```diff
@@ -14,7 +14,7 @@ services:
       DASH_DEBUG_MODE: "True"
       DASH_SILENCE_ROUTES_LOGGING: "False"
       DASH_SERVER_PORT: 8050
-      DB_HOST: db
+      DB_HOST: mongodb://db/openpath_prod_ca_ebike
```
Comment (JGreenlee) [inline docker-compose-dev.yml:17]: Was this change intentional? Shouldn't the default still be `Stage_database` ? @shankari @TeachMeTW

Comment (JGreenlee) [reply docker-compose-dev.yml:17]: Ah, I see this gets overwritten in the new version of the script so this value is arbitrary.

I still think the placeholder should be something other than `openpath_prod_ca_ebike` to prevent confusion

Comment (shankari) [reply docker-compose-dev.yml:17]: Ah good point. I had suggested that it be a placeholder like `REPLACE_ME_DB`, but didn't validate that it actually changed to be that. Agree that this should be changed to a placeholder in cleanup.

Comment (TeachMeTW) [reply docker-compose-dev.yml:17]: @shankari @JGreenlee It was changed to a placeholder but due to running the script, it changed once more (basically everytime you run the script the db location changes)

We could fix this in 2 ways I believe:

1. .gitignore the yaml from now on
2. Modify the script to revert the modification after launching compose?

Comment (shankari) [reply docker-compose-dev.yml:17]: I would suggest `.gitignore` and add a section to the `README` highlighting that if anybody does change it, they should use `git add -f <filename>`

That optimizes for the common case (people using the docker-compose) and drops the burden of doing extra work on the uncommon case (people who are modifying it) and who are also more likely to understand and notice it.

Comment (TeachMeTW) [reply docker-compose-dev.yml:17]: @shankari would you prefer a new pull request or directly change the repo since this PR was already merged?

Comment (shankari) [reply docker-compose-dev.yml:17]: @TeachMeTW you can't directly change the repo; you don't have write access to it. It has to be a PR 😄

Comment (TeachMeTW) [reply docker-compose-dev.yml:17]: @shankari moved to #123

**Comment by shankari** [comment]
@TeachMeTW the changes in this PR don't seem to have "stuck" to master.
Looking at the commit log, I think that they may have inadvertently gotten overwritten as part of some rebasing
the only commit I see to that file is one from Aug 2023
https://github.com/e-mission/op-admin-dashboard/commits/master/docker/load_mongodump.sh

Can you please restore the changes?

**Comment by TeachMeTW** [comment]
> @TeachMeTW the changes in this PR don't seem to have "stuck" to master. Looking at the commit log, I think that they may have inadvertently gotten overwritten as part of some rebasing to clean up the commit history.
> 
> the only commit I see to that file is one from Aug 2023 https://github.com/e-mission/op-admin-dashboard/commits/master/docker/load_mongodump.sh
> 
> Can you please restore the changes?

@shankari addressed in https://github.com/e-mission/op-admin-dashboard/pull/158
